src/librustc_typeck/astconv.rs

```diff
@@ -1011,7 +1011,7 @@ impl<'o, 'gcx: 'tcx, 'tcx> dyn AstConv<'gcx, 'tcx> + 'o {
 
         if regular_traits.is_empty() && auto_traits.is_empty() {
             span_err!(tcx.sess, span, E0224,
-                "at least one non-builtin trait is required for an object type");
+                "at least one non-builtin trait, outside of &send, is required for an object type");
```

Comment:
`&dyn Sync` works fine also.

The check above makes sure that there's at least one [regular trait | auto trait].

I think it would be best to say:

```
at least one trait is required for an object type
```